### PR TITLE
RHINENG-26412: fixed joi validation schemas and added edge case tests

### PR DESCRIPTION
**Pull request by rexwhite**
## Summary by Sourcery

Relax Joi validation of remediation issue IDs and expand test coverage for edge cases in patch and other handlers.

Bug Fixes:
- Allow issue IDs containing plus signs to pass validation for patch messages.
- Permit empty issues arrays across patch, advisor, compliance, and vulnerability handlers.
- Increase the maximum allowed issues per message to align with production payload sizes and enforce rejection when over the limit.
- Ensure issue IDs are validated against a 500-character length limit, rejecting longer values.
- Reject invalid host UUID formats such as UUID v1 in patch messages.

Tests:
- Add patch handler tests for empty issue arrays, large issue counts, issue ID length boundaries, UUID format validation, and plus sign handling in versions.
- Add advisor, compliance, and vulnerability handler tests to confirm empty issues arrays are accepted.

### CODE REVIEW

File: src/handlers/patch/patch.unit.ts [hunk 1]
```diff
@@ -64,4 +64,137 @@ describe('patch handler unit tests', function () {
         await handler(message);
         patchUpdateErrorParse.callCount.should.equal(1);
     });
+
```
Comment (marleystipich2) [inline src/handlers/patch/patch.unit.ts:67]: I guess we could include these extra tests in the other unit tests too 🤷‍♀️  Like compliance, vulnerabilities and advisor tests just in case

**Review by marleystipich2 — APPROVED** [review]
LGTM! I left one comment about the tests but just a suggestion 🤷‍♀️

File: src/handlers/patch/patch.unit.ts (hunk 1 above)
Comment (rexwhite) [reply src/handlers/patch/patch.unit.ts:67]: Good point - I didn't notice that Claude skipped those 🙄
